competition/services/database/gorm.go

```diff
@@ -11,21 +13,30 @@ import (
 	"arkavidia-backend-8.0/competition/models"
 )
 
-var currentDB *gorm.DB = nil
-
-func Init() *gorm.DB {
-	config := databaseConfig.GetDatabaseConfig()
-	dsn := fmt.Sprintf("host=%s user=%s password=%s dbname=%s port=%d", config.Host, config.User, config.Password, config.DBName, config.Port)
-
-	db, err := gorm.Open(postgres.Open(dsn), &gorm.Config{
-		Logger: logger.Default.LogMode(logger.Info),
-	})
-	if err != nil {
-		panic(err)
-	}
+type Database struct {
+	connection *gorm.DB
+	once       sync.Once
+}
 
-	// Create Type
-	if err := db.Exec(`
+// Private
+func (database *Database) lazyInit() {
+	database.once.Do(func() {
+		config := databaseConfig.Config.GetMetadata()
+		dsn := fmt.Sprintf("host=%s user=%s password=%s dbname=%s port=%d", config.Host, config.User, config.Password, config.DBName, config.Port)
+
+		db, err := gorm.Open(postgres.Open(dsn), &gorm.Config{
+			Logger: logger.Default.LogMode(logger.Info),
```

Comment:
Try https://github.com/sirupsen/logrus for logging, its quite amazing yes

Reply:
not only for the db but for general purpose

Reply:
Sip nanti dicoba geming :DD